### ISSUE TITLE
`dockerfile` service value documentation wrongly states it doesn't work with `image`

### ISSUE THREAD

**Issue by triccardi-systran** [docker/compose#3019]
[`dockerfile` doc](https://docs.docker.com/compose/compose-file/#dockerfile) hasn't been updated with Compose File v2 format:

> Using dockerfile together with image is not allowed. Attempting to do so results in an error.

This is wrong for v2 format, it does work:
```yaml
version: '2'
services:
  app:
    image: redis
    build:
      dockerfile: Dockerfile
      context: .
```

```shell
$ docker-compose config
networks: {}
services:
  test:
    build:
      context: /xxx
      dockerfile: Dockerfile
    image: redis
version: '2.0'
volumes: {}

```

**Comment by aanand** [comment]
It has been updated. That sentence appears beneath this one:

> Note: In the version 1 file format, dockerfile is different in two ways:

However, the formatting is strange and makes it look like those two statements are unconnected, so it's not as clear as it could be.

@moxiegirl @londoncalling: this looks like a CSS issue. Rather than adding a single left border to the enclosing `blockquote`, we're adding one to every `p` tag, resulting in multiple borders at different levels of indentation:

<img width="639" alt="screen shot 2016-02-25 at 14 18 57" src="https://cloud.githubusercontent.com/assets/1062/13336429/c0d6009e-dbca-11e5-9a60-7270057b92c1.png">

Can this be changed? If not, how should we change the markup/wording to make this section clearer?

**Comment by londoncalling** [comment]
@aanand yes, good call, I can make this change to some formatting that works better. I'll have a look now.

**Comment by aanand** [comment]
Closed by #3023.
